### PR TITLE
composite checkout: Support Ebanx TEF payments

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* Support bank transfer payments via Ebanx in composite checkout

To do:
- [x] Hook up submit function
- [x] Make sure validation and error messages work
- [x] Clean up styles of fields (done in #44806)
- [x] Figure out why submission doesn't succeed
- [x] Fix logo and styling of collapsed payment method

#### Testing instructions

See D47729-code for forcing this payment method from the sandbox, and make sure you have `BILLING_SANDBOX` enabled to use the Ebanx sandbox environment. Test customer data is available at https://developer.ebanx.com/docs/resources/testCustomerData/#brazilian-test-data. (Note that CEP is the postal code, and CPF is the tax ID.) Also be aware that it is normal for bank transfer payments (event test payments) to take a while; whatever action you take on the redirect page, it may look like the provisioning page is stuck - just wait a few minutes.

* Enter checkout somehow (signup, new purchase, renewal) with some combination of products (plan only, plan+domain, extra domain only, etc.)
* Proceed through checkout to the payment method step. Check that the bank transfer method is available (D47729-code will force this from your sandbox)
* Select this method and jerk test the crap out of it. Particularly important flows include:
    * Try to submit with incomplete form data (missing tax ID, missing parts of the address, missing the name, missing the bank)
    * Try to submit with a contact country code that isn't BR. (There is no country code in the payment method itself, since it has to be BR, so we send the contact country code to Ebanx. If it is not BR, check that we show a useful error message.)
    * Submit with correct information, but at the redirect page, do not authorize the txn. Check that we show a useful error message and that checkout doesn't get stuck.
    * Submit with correct information and authorize the charge at the redirect page. Check that we generate a receipt in payments admin, and that the metadata on the receipt looks right (payment partner should be Ebanx, and the "card last 4 and expiry" will be all zeros because that doesn't exist for bank transfers.)

image

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/payment-method-helpers.js [hunk 1]
```diff
@@ -138,6 +138,7 @@ export async function submitEbanxCardTransaction( transactionData, submit ) {
 	return submit( formattedTransactionData );
 }
 
+// NOTE: not really stripe specific
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/payment-method-helpers.js:141]: Maybe we should just rename this function to `submitRedirectTransaction`?

File: client/my-sites/checkout/composite-checkout/payment-method-helpers.js [hunk 2]
```diff
@@ -156,6 +157,15 @@ export async function submitStripeRedirectTransaction( paymentMethodId, transact
 	return submit( formattedTransactionData );
 }
 
+export async function submitEbanxTefTransaction( transactionData, submit ) {
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/payment-method-helpers.js:160]: I don't think this function is used anywhere, right?

File: client/my-sites/checkout/composite-checkout/payment-methods/ebanx-tef.js [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+					getFieldValue={ getFieldValue }
+					handleFieldChange={ setFieldValue }
+					disableFields={ isDisabled }
+				/>
+			</div>
+		</EbanxTefFormWrapper>
+	);
+}
+
+// TODO: Merge this with payment-methods/credit-card/contact-fields.js
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/payment-methods/ebanx-tef.js:208]: Maybe we should make a new component inside `my-sites/checkout/composite-checkout/` somewhere that includes this wrapper. Like `CompositeCheckoutCountrySpecificPaymentFields` or similar.

File: client/my-sites/checkout/composite-checkout/payment-method-helpers.js (hunk 2 above)
Comment (nbloomf) [reply client/my-sites/checkout/composite-checkout/payment-method-helpers.js:160]: I rebased at some point and this got included in the rebased branch by mistake- it's gone now.

Comment (sirbrillig) [reply client/my-sites/checkout/composite-checkout/payment-method-helpers.js:159]: But... it's still here. 🤔

Comment (nbloomf) [reply client/my-sites/checkout/composite-checkout/payment-method-helpers.js:159]: Ah- I see what I did wrong. I `rg`d for the function name to find all the instances of it, but misspelled the name so there were no results. Meanwhile I really did have to remove some rebase gunk and assumed this was that function. Thanks for checking again :)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4223083

Thank you @nbloomf for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4223083).

**Pull request merged**